src/Services/InventoryUpdateService.php

```diff
@@ -37,319 +54,524 @@ class InventoryUpdateService
+  private function getNormalizedStockBuffer()
   {
     $stockBuffer = null;
-    if (isset($configHelper)) {
-      $stockBuffer = $configHelper->getStockBufferValue();
-    }
+    $stockBuffer = $this->configHelper->getStockBufferValue();
+
 
     if (isset($stockBuffer)) {
       if ($stockBuffer >= 0) {
         return $stockBuffer;
       }
 
       // invalid value for buffer
-      if (isset($loggerContract)) {
-        $loggerContract->warning(
-          TranslationHelper::getLoggerKey(self::LOG_KEY_INVALID_STOCK_BUFFER),
-          [
-            'additionalInfo' => ['stockBuffer' => $stockBuffer],
-            'method' => __METHOD__
-          ]
-        );
-      }
+      $this->logger->warning(
+        TranslationHelper::getLoggerKey(self::LOG_KEY_INVALID_STOCK_BUFFER),
+        [
+          'additionalInfo' => [
+            'stockBuffer' => json_encode($stockBuffer)
+          ],
+          'method' => __METHOD__
+        ]
+      );
     }
 
     return 0;
   }
+
+  /**
+   * Check if the state of the Inventory Service has been "running" for more than the maximum allotted time
+   * This functionality was extracted from the old UpdateFullInventoryStatusCron
+   * @param bool $full check full inventory service instead of partial
+   * @return bool
+   */
+  private function serviceHasBeenRunningTooLong(bool $full): bool
+  {
+
+    if (!$this->statusService->isInventoryRunning($full)) {
+      return false;
+    }
+
+    $maxTime = $full ? self::MAX_INVENTORY_TIME_FULL : self::MAX_INVENTORY_TIME_PARTIAL;
+    $logKey = $full ? self::LOG_KEY_LONG_RUN_FULL : self::LOG_KEY_LONG_RUN_PARTIAL;
+
+    $lastStart = $this->statusService->getLastAttemptTime($full);
+    if (isset($lastStart) && !empty($lastStart) && (time() - strtotime($lastStart)) > $maxTime) {
+
+      $this->logger->warning(TranslationHelper::getLoggerKey($logKey), [
+        'additionalInfo' => ['startedOn' => $lastStart, 'maximumTime' => $maxTime],
+        'method' => __METHOD__
+      ]);
+      return true;
+    }
+
+    return false;
+  }
+
+  /**
+   * Get the php time (in seconds) for the start of a partial sync
+   *
+   * @param ExternalLogs $externalLogs
+   *
+   * @return int
+   */
+  private function getStartOfDeltaSyncWindow(ExternalLogs $externalLogs = null): int
+  {
+    $windowStart = 0;
+
+    $lastWindowEnd = $this->statusService->getLastSuccessfulAttemptTime(false);
+    $lastGoodFullStart = $this->statusService->getLastSuccessfulAttemptTime(true);
+
+    if (isset($lastWindowEnd) && !empty($lastWindowEnd)) {
+      // new window should be directly after the previous window
+      $windowStart = strtotime($lastWindowEnd);
+    }
+
+    if (isset($lastGoodFullStart) && !empty($lastGoodFullStart)) {
+      $numericTimeForFullSync = strtotime($lastGoodFullStart);
+      if ($windowStart > 0 && $numericTimeForFullSync < $windowStart) {
+        // full sync happened more recently than partial sync - use that time.
```

Comment:
Am I misunderstanding this or doesn't $numericTimeForFullSync < $windowStart imply that the partial sync happened more recently?

Reply:
hm yeah this may be backwards.